src/world/world_server.cpp

```diff
@@ -21,12 +21,13 @@ along with this program.  If not, see http://www.gnu.org/licenses/
 #include "world_server.h"
 
 #include "common/application.h"
-#include "common/console_service.h"
 #include "common/logging.h"
+#include "common/vana_time.h"
 
 namespace
 {
-    std::unique_ptr<SqlConnection> sql;
+    std::unique_ptr<SqlConnection>  sql;
+    std::unique_ptr<ConquestSystem> conquest;
```

Comment:
There's already a std::shared_ptr<ConquestSystem> conquestSystem; in worldServer members?

Reply:
We need to expose world in the namespace to then right?

Reply:
This pointer is the one in world, assigned on construction, just like sql

Reply:
An anonymous namespace is used to hide local objects/data within a given compilation unit - stopping them from being extern'd from elsewhere. You can _technically_ do either way, but both is messy and confusing. The Conquest System should be a unique ptr member of the class, it doesn't need to be hidden. The local SQL ptr makes more sense in the anonymous namespace because it is explicitly _not_ for sharing, since most systems on world server will have their own and everything is threaded

Reply:
```cpp
WorldServer::WorldServer(int argc, char** argv)
: Application("world", argc, argv)
, conquestSystem()
, httpServer(std::make_unique<HTTPServer>())
, messageServer(std::make_unique<message_server_wrapper_t>(std::ref(m_RequestExit)))
{
    // Anonymous namespace preparation
    sql      = std::make_unique<SqlConnection>();
    conquest = std::make_unique<ConquestSystem>();
```

This means you have 2x conquest systems now. `, conquestSystem()` is constructing the WorldServer::conquestSystem member, and `conquest = std::make_unique<ConquestSystem>();` is constructing the anonymous namespace member conquest in this compilation unit.

Reply:
Oh right yes. I did that intentionally because conquestSystem is stateless and I did not know a better way to share the conquest system. However i'd prefer being able to use the world system instance from world_server.cpp, but how do we do that? (i.e for the tally tick logic)

Reply:
Done. The same shared_ptr can be used for both